browser/src/Services/InputManager.ts

```diff
@@ -0,0 +1,84 @@
+
+import { commandManager } from "./CommandManager"
+
+export type ActionFunction = () => boolean
+
+export type ActionOrCommand = string | ActionFunction
+
+export type FilterFunction = () => boolean
+
+export interface KeyBinding {
+    action: ActionOrCommand
+    filter?: FilterFunction
+}
+
+export interface KeyBindingMap {
+    [key: string]: KeyBinding[]
+}
+
+export class InputManager implements Oni.InputManager {
+
+    private _boundKeys: KeyBindingMap = {}
+
+    /**
+     * API Methods
+     */
+    public bind(keyChord: string, action: ActionOrCommand, filterFunction?: () => boolean) {
+        // tslint:disable-line no-empty-block
+
+        // TODO: add to existing binding
```

Comment:
Ya, basically if I do something like:

```
const isVisualMode = () => Oni.editors.activeEditor.mode === "visual"
const isNormalMode = () => Oni.editors.activeEditor.mode === "normal"
Oni.input.bind("<C-c>", "someCommand",  isVisualMode)
Oni.input.bind("<C-c>", "someOtherCommand", isNormalMode)
```

Only the last binding would work right now, since we clear out old bindings. That's the difference I had between `bind` and `rebind` - `bind` preserves existing bindings, but `rebind` clears them all.

Perhaps it's better just to have `bind` and `unbind`, and if it ends up being a common pattern to call `unbind` + `bind`, we can bring `rebind` back as a convenience.